### PR TITLE
meson: Enable big objects support

### PR DESCRIPTION
**Pull request by chouquette**
Fix cross compilation to win64

### CODE REVIEW

**Comment by chouquette** [comment]
For reference, this is what this MR fixes: https://code.videolan.org/videolan/vlc/-/jobs/903124

File: meson.build [hunk 1]
```diff
@@ -57,6 +57,10 @@ if host_machine.cpu_family() == 'arm' and cpp.alignment('struct { char c; }') !=
   endif
 endif
 
+if host_machine.system() == 'windows' and cpp.get_id() != 'msvc'
+  add_project_arguments(['-Wa,-mbig-obj'], language : 'cpp')
+endif
```
Comment (khaledhosny) [inline meson.build:62]: Why is this only for compilers other than MSVC? Also the check seems to assume any compiler that is not MSVC is GCC-like, not sure if this is a right assumption to make.

Comment (chouquette) [reply meson.build:62]: Different than MSVC, but only when compiling for Windows. In my experience that usually means gcc/llvm, but that's a valid point.

I can refactor the change to add it whenever host_machine is windows, and if the compiler supports it, if that's ok with you

Comment (khaledhosny) [reply meson.build:62]: Yes, please.

Comment (chouquette) [reply meson.build:64]: Done

**Pull request merged**